app/controllers/api/addresses_controller.rb

```diff
@@ -7,8 +7,7 @@ class Api::AddressesController < Api::BaseController
   filter_resources_by :account_id
 
   def resource
-    @address = account.address if !super && account
-    @address
+    @address ||= account.try(:address) || super
```

Comment:
This is semantically different from the old behavior, but I don't know enough to know whether it matters. Just calling it out, but will trust if you tell me it's right.